lib/settings/settings.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const assert  = require('assert')
+
+const KEY = Symbol()
+const POLL_INTERVAL = Symbol()
+
+module.exports = (config, mc, log) => {
+
+
+  // A sentinel error for signaling that the result was invalid/missing,
+  // and we should try to push our own representation to memcached.
+  class PushOnMissing extends Error {}
```

Comment:
api nit: something about the name `PushOnMissing` here doesn't feel right to me.  The correspondingly-named argument to `refresh` means "you should push if the item turns out to be missing", the situation here is "the item was actually missing".  Would naming this something like `ItemMissing` make things clearer?

I don't feel super strongly about it though.